From eddcb2a5cd9579e3b70b6aad56f1d5e89612f83a Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Fri, 20 Nov 2020 14:11:04 +0200
Subject: [PATCH] Allow nullable head parameter in stash_if_needed()

Newer versions of Vala perform parameter checks in async methods, so
Gitg.ActionSupport.stash_if_needed's head param should be Gitg.Ref?.

Related Vala issue: https://gitlab.gnome.org/GNOME/vala/-/issues/299

Fixes the following assertion failure:

  (tests/gitg/tests-gitg:2977): gitg-CRITICAL **: 07:44:32.083:
    gitg_action_support_stash_if_needed: assertion 'head != NULL' failed
---
 gitg/gitg-action-support.vala | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gitg/gitg-action-support.vala b/gitg/gitg-action-support.vala
index 6eb7ad8d..d864fffc 100644
--- a/gitg/gitg-action-support.vala
+++ b/gitg/gitg-action-support.vala
@@ -122,7 +122,7 @@ public class ActionSupport : Object
 		return head != null;
 	}
 
-	public async bool stash_if_needed(SimpleNotification notification, Gitg.Ref head)
+	public async bool stash_if_needed(SimpleNotification notification, Gitg.Ref? head)
 	{
 		// Offer to stash if there are any local changes
 		if ((yield working_directory_dirty()))
-- 
GitLab

